src/fides/api/api/v1/endpoints/dataset_config_endpoints.py

```diff
@@ -254,59 +200,23 @@ def patch_dataset_configs(
 
     The CtlDataset contents are retrieved for extra validation before linking this
     to the DatasetConfig.
-
     """
-    created_or_updated: List[Dataset] = []
-    failed: List[BulkUpdateFailed] = []
-    logger.info("Starting bulk upsert for {} Dataset Configs", len(dataset_pairs))
-
-    for dataset_pair in dataset_pairs:
-        logger.info(
-            "Finding ctl_dataset with key '{}'", dataset_pair.ctl_dataset_fides_key
+    try:
+        return dataset_config_service.bulk_create_or_update_dataset_configs(
+            connection_config,
+            dataset_pairs,
         )
-        ctl_dataset: CtlDataset = (
-            db.query(CtlDataset)
-            .filter_by(fides_key=dataset_pair.ctl_dataset_fides_key)
-            .first()
+    except DatasetNotFoundException as e:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=str(e),
         )
-        if not ctl_dataset:
-            raise HTTPException(
-                status_code=HTTP_404_NOT_FOUND,
-                detail=f"No ctl dataset with key '{dataset_pair.ctl_dataset_fides_key}'",
-            )
-
-        try:
-            fetched_dataset: Dataset = Dataset.model_validate(ctl_dataset)
-        except PydanticValidationError as e:
-            raise HTTPException(
-                status_code=HTTP_422_UNPROCESSABLE_ENTITY,
-                detail=jsonable_encoder(
-                    e.errors(include_url=False, include_input=False)
-                ),
-            )
-        validate_data_categories(fetched_dataset, db)
-
-        data = {
-            "connection_config_id": connection_config.id,
-            "fides_key": dataset_pair.fides_key,
-            "ctl_dataset_id": ctl_dataset.id,
-        }
-
-        create_or_update_dataset(
-            connection_config,
-            created_or_updated,
-            data,
-            fetched_dataset,
-            db,
-            failed,
-            DatasetConfig.create_or_update,
```

Comment:
Got rid of this function in favor of a simplified `create_or_update_dataset_config` and `bulk_create_or_update_dataset_configs` functions